### ISSUE TITLE
Enhancement: Enforce Minimum TLS Version

### ISSUE THREAD

**Issue by Hacks4Snacks** [senthilrch/kube-fledged#231]
The webhook server does not enforce a minimum TLS version of 1.2. TLS 1.0 and 1.1 are outdated and have known vulnerabilities.

**Comment by Hacks4Snacks** [comment]
If accepted I am more than happy to get a PR submitted.

**Comment by sshiba** [comment]
I have a requirement that requires the ability to enforce a minimum TLS version of 1.2 and it will benefit if this project is accepted.

**Comment by samirgrl** [comment]
Good catch @Hacks4Snacks. Enforcing a minimum TLS version (TLS 1.2+) on the webhook server is a reasonable security hardening step, especially for clusters that need to satisfy compliance requirements (SOC 2, PCI, FedRAMP, etc.). The Go TLS library makes this straightforward — `tls.Config{MinVersion: tls.VersionTLS12}`.

I've opened issue #242 proposing to take over as maintainer. Security hardening like this is something I'd want to act on — it has low implementation risk and high compliance value. Happy to help draft the fix if there's interest.
